src/handlers/diff_header.rs

```diff
@@ -154,7 +167,43 @@ impl<'a> StateMachine<'a> {
             self.config,
         );
         // FIXME: no support for 'raw'
-        write_generic_diff_header_header_line(&line, &line, &mut self.painter, self.config)
+        write_generic_diff_header_header_line(
+            &line,
+            &line,
+            &mut self.painter,
+            &mut self.mode_info,
+            self.config,
+        )
+    }
+
+    pub fn handle_pending_mode_line_with_diff_name(&mut self) {
+        if !self.mode_info.is_empty() {
+            let format_label = |label: &str| {
+                if !label.is_empty() {
+                    format!("{} ", label)
+                } else {
+                    "".to_string()
+                }
+            };
+            let format_file = |file| {
+                if self.config.hyperlinks {
+                    features::hyperlinks::format_osc8_file_hyperlink(file, None, file, self.config)
+                } else {
+                    Cow::from(file)
+                }
+            };
+            let label = format_label(&self.config.file_modified_label);
+            let name = get_repeated_file_path_from_diff_line(&self.diff_line)
+                .unwrap_or_else(|| "".to_string());
+            let line = format!("{}{}", label, format_file(&name));
+            let _got = write_generic_diff_header_header_line(
+                &line,
+                &line,
+                &mut self.painter,
+                &mut self.mode_info,
+                self.config,
+            );
```

Comment:
Rather than assigning to an unused variable to avoid handling the case where this returns an error, I believe the more correct implementation would be along the lines of the diff below. I.e. change this function to return a `Result` enum, and propagate the error handling upwards. This also fits the pattern in `delta::consume` where functions named `handle_foo` all return a `Result` (although this one is slightly different as it doesn't return a bool saying whether it has handled a line).  

```diff
commit 212931987f591ae70d4da594024c46eae3b105f5
Author: Dan Davison <dandavison7@gmail.com>
Date:   Mon Jan 3 16:13:55 2022 -0500

    WIP

diff --git a/src/delta.rs b/src/delta.rs
index 1e6fd84..7ba2997 100644
--- a/src/delta.rs
+++ b/src/delta.rs
@@ -173,7 +173,7 @@ impl<'a> StateMachine<'a> {
                 || self.emit_line_unchanged()?;
         }
 
-        self.handle_pending_mode_line_with_diff_name();
+        self.handle_pending_mode_line_with_diff_name()?;
         self.painter.paint_buffered_minus_and_plus_lines();
         self.painter.emit()?;
         Ok(())
diff --git a/src/handlers/diff_header.rs b/src/handlers/diff_header.rs
index 61b2aa9..ea1c4dd 100644
--- a/src/handlers/diff_header.rs
+++ b/src/handlers/diff_header.rs
@@ -176,7 +176,7 @@ impl<'a> StateMachine<'a> {
         )
     }
 
-    pub fn handle_pending_mode_line_with_diff_name(&mut self) {
+    pub fn handle_pending_mode_line_with_diff_name(&mut self) -> std::io::Result<()> {
         if !self.mode_info.is_empty() {
             let format_label = |label: &str| {
                 if !label.is_empty() {
@@ -196,13 +196,15 @@ impl<'a> StateMachine<'a> {
             let name = get_repeated_file_path_from_diff_line(&self.diff_line)
                 .unwrap_or_else(|| "".to_string());
             let line = format!("{}{}", label, format_file(&name));
-            let _got = write_generic_diff_header_header_line(
+            write_generic_diff_header_header_line(
                 &line,
                 &line,
                 &mut self.painter,
                 &mut self.mode_info,
                 self.config,
-            );
+            )
+        } else {
+            Ok(())
         }
     }
 }
diff --git a/src/handlers/diff_header_diff.rs b/src/handlers/diff_header_diff.rs
index c2d7a86..fdd2d39 100644
--- a/src/handlers/diff_header_diff.rs
+++ b/src/handlers/diff_header_diff.rs
@@ -22,7 +22,7 @@ impl<'a> StateMachine<'a> {
             } else {
                 State::DiffHeader(DiffType::Unified)
             };
-        self.handle_pending_mode_line_with_diff_name();
+        self.handle_pending_mode_line_with_diff_name()?;
         self.handled_diff_header_header_line_file_pair = None;
         self.diff_line = self.line.clone();
         if !self.should_skip_line() {
```

Reply:
Sounds good!